backend/question-service/server/app/routers/admin.py

```diff
@@ -0,0 +1,94 @@
+from fastapi import HTTPException, Query, APIRouter
+from typing import Optional
+import sys
+
+sys.path.append("..")
+from core.database import collection
+from utils.jsend import JSendStatus, jsend_response
+
+
+admin_router = APIRouter(
+    prefix="/admin",
+    tags=["Admin"],
+)
+
+
+@admin_router.post("/questions")
+async def create_question(question: dict):
+    """Creates a new question in the database"""
+
+    # Check if id is provided. If not, auto-increment from the last id
+    if "id" not in question:
+        last_question = collection.find_one(sort=[("id", -1)])
+        if last_question:
+            question["id"] = last_question["id"] + 1
+        else:
+            question["id"] = 1
+
+    # Check if question with the same id already exists
+    if collection.find_one({"id": question["id"]}):
+        raise HTTPException(
+            status_code=400, detail="Question with the same id already exists.")
+    result = collection.insert_one(question)
+
+    question["_id"] = str(question["_id"])
+
+    return jsend_response(JSendStatus.SUCCESS, {"inserted_data": question})
+
+
+@admin_router.get("/questions")
+async def get_questions(
+    skip: int = Query(
+        0, alias="page", description="Page number for pagination. Starts from 0."),
+    limit: int = Query(
+        10, description="Number of items to retrieve per page."),
```

Comment:
So are `skip` and `limit` mandatory params? Does `Query(0, ...` and `Query(10, ...` represent default values?

Reply:
They represent default values (ie. if none is provided). So a request like `GET /admin/questions` with no query parameters will still work but with the default parameters.

Reply:
Added pagination details in response body https://github.com/CS3219-AY2324S1/ay2324s1-course-assessment-g03/commit/6617a42ae09dab7ee6beaa7d478b25451f56c033